includes/Linker.php

```diff
@@ -464,7 +464,11 @@ static function normaliseSpecialPage( Title $title ) {
 	 */
 	private static function fnamePart( $url ) {
 		if (VignetteRequest::isVignetteUrl($url)) {
-			return VignetteRequest::getImageFilename($url);
+			try {
+				return VignetteRequest::getImageFilename($url);
+			} catch (Exception $e) {
+				// VignetteUrlToUrlGenerator logs the exact error
```

Comment:
Empty `catch` block looks a bit strange. If the error is logged in `VignetteUrlToUrlGenerator` class why do we throw an exception here?